main.c

```diff
@@ -57,6 +57,42 @@ int lenient_strcmp(char *a, char *b) {
 	}
 }
 
+#ifdef ENABLE_READYFD
+static void readyfd(void) {
+	int fd;
+	char *env = NULL, *endptr = NULL;
+
+	env = getenv("READY_FD");
+
+	if (!env || env[0] == '\0') {
+		swaylock_log(LOG_ERROR, "READY_FD not set or empty string");
+		exit(1);
+	}
+
+	if (unsetenv("READY_FD") != 0) {
+		swaylock_log(LOG_ERROR, "Could not unset READY_FD");
+		exit(1);
+	}
+
+	errno = 0;
+	fd = (int)strtol(env, &endptr, 10);
+
+	if (errno != 0 || endptr[0] != '\0') {
+		swaylock_log(LOG_ERROR, "Could not convert READY_FD to int");
+		exit(1);
```

Comment:
The spec says to print a warning in this case.

Reply:
Since the flag is explicitly requested, I think an error is more appropriate and an update to the spec is in order.